src/uct/cuda/base/cuda_iface.h

```diff
@@ -76,10 +76,22 @@ static UCS_F_ALWAYS_INLINE int uct_cuda_base_is_context_active()
 }
 
 
+static UCS_F_ALWAYS_INLINE int uct_cuda_base_is_context_valid(CUcontext ctx)
+{
+    unsigned version;
+    ucs_status_t status;
+
+    /* Check if CUDA context is valid by running a dummy operation on it */
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetApiVersion(ctx, &version));
+    return (status == UCS_OK);
+}
+
+
 static UCS_F_ALWAYS_INLINE int uct_cuda_base_context_match(CUcontext ctx1,
                                                            CUcontext ctx2)
 {
-    return ((ctx1 != NULL) && (ctx1 == ctx2));
+    return ((ctx1 != NULL) && (ctx1 == ctx2) &&
+            uct_cuda_base_is_context_valid(ctx1));
```

Comment:
now it will be called from multiple places, including progress, do we really want it?

Reply:
I checked that, `uct_cuda_base_context_match` is called from 2 places:
1. UCS_CLASS_CLEANUP_FUNC (uct_cuda_ipc_iface_t, uct_cuda_copy_iface_t)
2. uct_cuda_copy_event_desc_cleanup, uct_cuda_ipc_event_desc_cleanup
These functions are invoked on memory pool chunks cleanup, when we deallocate the chunk. Normally this happens at UCX destroy. So it should not add any overhead to the hot path
Or I missed something?

Reply:
right, i mixed with other function